src/transforms/sample/transform.rs

```diff
@@ -12,15 +12,44 @@ use crate::{
 use vector_lib::lookup::lookup_v2::OptionalValuePath;
 use vector_lib::lookup::OwnedTargetPath;
 
+/// Exists only for backwards compatability purposes so that the value of sample_rate_key is
+/// consistent after the internal implementation of the Sample class was modified to work in terms
+/// of percentages
+#[derive(Clone, Copy, Debug)]
+pub enum SampleRate {
+    OneOverN(u64),
+    Percentage(f32),
```

Comment:
Since the other arm is a `u64`, there is no benefit and potential precision loss to storing this as a `f32` instead of `f64` as well.

Reply:
There would still be precision loss though with `f64` right? Just a much smaller possible set of numbers. Will modify and leave the comment in though